rfc-175-frontend-fewer-apps.md

```diff
@@ -0,0 +1,254 @@
+---
+status: proposed
+implementation: proposed
+status_last_reviewed:
+---
+
+# Consolidate Frontend Content-Rendering Apps
+
+## Summary
+
+This is a refinement of the suggestions in [RFC-174] about reducing the number of frontend rendering applications, reducing its scope to the applications that deal exclusively with rendering content items. By dealing only with the four applications that exclusively render content items, we improve standardisation and reduce maintenance costs, and the work is worthwhile even if we can only achieve sections of it. Because a large part of the problem and a lot of the solution remains the same, we have divided this RFC into two parts - an initial section that describes how we differ from [RFC-174], and a more verbose section detailing the approach. If you have already read or commented on [RFC-174], you can read the [How this proposal differs](#how-this-proposal-differs) section and skip the rest of the RFC. If you are new to this proposal, feel free to skip to the [Full Exploration of the Problem](#full-exploration) section.
+
+## <a name="how-this-proposal-differs"></a>How this proposal differs from [RFC-174]
+
+### Reduced Scope
+
+We have reduced the scope of the RFC to just consider the 4 pure content rendering apps:
+
+1. [collections]
+2. [finder-frontend]
+3. [frontend]
+4. [government-frontend]
+
+### Removed references to retiring router/router-api
+
+Retiring router/router-api in [RFC-174] relied on us dealing with [smart-answers], a problem worthy of its own RFC. With the reduced scope we can’t easily retire the router and router-api, although it does still simplify other avenues for retiring router that we can potentially explore later.
+
+### Removed references to retiring static
+
+Again, retiring [static] by merging all the apps relies on all apps being merged, something that is far enough in the future that we’ve decided to explore other options for retiring [static]
+
+## <a name="full-exploration"></a>Full Exploration of the Problem
+
+### Applications overview
+GOV.UK is served by multiple rendering applications (apps). Each app serves different groups of pages within the website.
+
+There are currently 8 rendering apps for GOV.UK that are supported and maintained by GOV.UK teams, 4 of which are almost exclusively concerned with rendering content:
+
+1. [collections]
```

Comment:
I don't think there's any reason this couldn't be merged to frontend so it's a bit out of scope of the RFC really, but do you see the merged frontend app as the long term home of the organisations API, which currently resides in collections?

Reply:
What currently uses that API?

Reply:
[A fair few things](https://github.com/search?q=org%3Aalphagov+GdsApi%3A%3AOrganisations&type=code), although I think a lot of them are no longer active. Manuals Publisher and Short URL Manager are though

Reply:
Ah, I see it's here: https://github.com/alphagov/collections/blob/main/docs/api.md - mainly internal apps. Good question! It's probably not a terrible place for it to be, given that it's publicly available information and other people _might_ potentially use it.

Reply:
In some ways, the merged app would become a more appropriate place for it, because it's somewhat like the search results in Finder Frontend. Perhaps Search API is a natural home? I think it's a tricky question, because there's no particular good home for it, so maybe for the moment it makes as much sense to keep it here as anywhere. But when the routes are moved, we'll consider if there's a better place - we don't want to treat the grouping of the routes in the four current apps as handed down to us on stone tablets.